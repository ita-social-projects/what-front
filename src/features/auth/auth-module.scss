.wrapper {
  position: absolute;
  width: 100%;
  min-height: 100%;  
  background-color: #01073A;  
}

.auth-container {
  background-color: #fff;
  margin-top: 40vh;  
}
