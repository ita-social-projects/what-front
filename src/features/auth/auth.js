import React from 'react';
import styles from './auth.scss';

export const Auth = () => {  
  return (
    <div className={styles.wrapper}>    
      <div className="container">
        <div className="row justify-content-center">
          <div className="col-lg-4 col-md-6 col-sm-8">            
            <div className={styles.form}>
              <form className="p-3">
                <div className="form-group">
                  <input type="email" className="form-control" id="auth-form__email" placeholder="Email address" />
                </div>
                <div className="form-group">
                  <input type="password" className="form-control" id="auth-form__password" placeholder="Password" />
                </div>
                <button type="submit" className="btn btn-block btn-warning">Sign in</button>
                <div className="text-center mt-3">
                  <p>Don't have an account? <a href="#" className={styles.link}>Registration</a></p>                  
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}