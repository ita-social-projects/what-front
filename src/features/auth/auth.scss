.wrapper {
  position: absolute;
  width: 100%;
  min-height: 100%;  
  background-color: #01073A;  
}

.form {
  background-color: #fff;
  margin-top: 40vh;
  border-radius: 3px;;
}

.link { 

  &:hover {
    text-decoration: none;    
  }
}
