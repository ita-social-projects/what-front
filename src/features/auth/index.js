export { Auth } from './auth.js';