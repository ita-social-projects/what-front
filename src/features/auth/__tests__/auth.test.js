import React from 'react';
import { useSelector } from 'react-redux';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { useActions } from '@/shared/index.js';
import { login } from '@models/index.js';
import { Auth } from '../auth.js';

jest.mock('react-redux', () => ({ useSelector: jest.fn().mockReturnValue({
  isLoading: false,
  error: '',
  loaded: true,
}) }));

jest.mock('@/shared/hooks', () => ({
  useActions: jest.fn().mockReturnValue(jest.fn()),
}));

describe('Auth', () => {
  let formValues;
  beforeEach(() => {
    formValues = {
      email: 'testmail@gmail.com',
      password: 'Testpassw!ord123',
    };
  });

  it('should render component', () => {
    const { container } = render(<Router><Auth /></Router>);
    const authContainer = container.getElementsByClassName('container');

    expect(authContainer).toMatchSnapshot();
  });

  it('should validate the form', async () => {
    const { getByText, getAllByText } = render(<Router><Auth /></Router>);

    const FORM_FIELD_AMOUNT = 2;
    const submitButton = getByText('Sign in');

    await waitFor(() => {
      fireEvent.click(submitButton);
    });

    const errors = getAllByText('This field is required');

    expect(errors).toHaveLength(FORM_FIELD_AMOUNT);
    expect(errors).toMatchSnapshot();
  });

  it('should handle error receiving', () => {
    useSelector.mockReturnValue({
      isLoading: false,
      error: 'Something went wrong',
      loaded: true,
    });

    const { getByText } = render(<Router><Auth /></Router>);

    const error = getByText('Something went wrong');
    expect(error).toBeInTheDocument();
  });

  it('should submit the form with correct schema', async () => {
    const { getByPlaceholderText, getByText } = render(<Router><Auth /></Router>);

    const email = getByPlaceholderText('Email address');
    const password = getByPlaceholderText('Password');
    const submitButton = getByText('Sign in');

    const signIn = useActions(login);

    fireEvent.change(email, {
      target: {
        value: formValues.email,
      },
    });

    fireEvent.change(password, {
      target: {
        value: formValues.password,
      },
    });

    await waitFor(() => {
      fireEvent.click(submitButton);
    });

    signIn({
      email: email.value,
      password: password.value,
    });

    expect(signIn).toHaveBeenCalledWith(formValues);
  });
});