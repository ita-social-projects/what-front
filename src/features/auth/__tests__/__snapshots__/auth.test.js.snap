// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render component 1`] = `
HTMLCollection [
  <div
    class="container"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-xl-6 col-md-8 col-sm-10 col-12"
      >
        <div
          class="card shadow"
        >
          <form
            action="#"
            class="p-3"
            novalidate=""
          >
            <h3
              class="text-center"
            >
              Sign in to WHAT
            </h3>
            <hr />
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email address
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="Email address"
                type="email"
                value=""
              />
              <p
                class="text-danger"
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="on"
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <p
                class="text-danger"
              />
            </div>
            <div
              class="d-flex justify-content-center"
            >
              <button
                class="btn undefined"
                type="submit"
              >
                Sign in
              </button>
            </div>
            
            <div
              class="d-flex justify-content-between mt-3"
            >
              <div
                class="text-center"
              >
                <p>
                  <a
                    href="/forgot-password"
                  >
                    Forgot Password?
                  </a>
                </p>
              </div>
              <div
                class="text-right"
              >
                <p>
                  Don't have an account? 
                  <a
                    href="/registration"
                  >
                    Registration
                  </a>
                </p>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Auth should validate the form 1`] = `
Array [
  <p
    class="text-danger"
  >
    This field is required
  </p>,
  <p
    class="text-danger"
  >
    This field is required
  </p>,
]
`;
